checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -354,9 +606,342 @@ protected boolean skipContractCheck(Tree tree, FlowExpressions.Receiver expr, Fl
                 }
             }
         } catch (FlowExpressionParseException e) {
-            // errors are reported at declaration site
+            checker.report(e.getResult(), tree);
         }
 
         return false;
     }
+
+    @Override
+    public boolean isValidUse(AnnotatedDeclaredType declarationType,
+            AnnotatedDeclaredType useType, Tree tree) {
+        declarationType.replaceAnnotation(GUARDEDBY);
```

Comment:
Do you know of a better solution to this problem?
If not, I'll add a comment to the code.